packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -1368,8 +1391,10 @@ class _DateRangePickerDialogState extends State<DateRangePickerDialog> with Rest
         );
```

Comment:
In the previous few lines, you can use the local `useMaterial3` instead of looking up in the theme. (Github won't let me comment on the correct lines).

Reply:
Addressed in https://github.com/flutter/flutter/pull/119033/commits/70d860d720f506a57c730749eaaba252e4cd8a6b